dvc/api.py

```diff
@@ -240,4 +243,6 @@ def _import_string(import_name):
 
 def _require_dvc(repo):
     if not isinstance(repo, Repo):
+        if os.path.exists(repo.url):
+            raise NotDvcRepoError(repo.url)
```

Comment:
The assumptions in tests were correct, only needed to add this.

Reply:
Is this correct (the given path is a git only repo)? @efiop 
```sh
➜ dvc get /tmp/tmp.Fn43F3mvVy foo --show-url   
ERROR: failed to show url - you are not inside of a dvc repository (checked up to mount point '/tmp')
```

Reply:
We probably need a different exception, one without `you are not inside of`.

Reply:
Okay, I'll make this change on a different PR. I will use the same exception, but somehow throw different message.

Reply:
@skshetry Could you please elaborate a bit why this `if` is needed? Also, why doesn't it raise `UrlNotDvcRepoError`?

Reply:
I did not want to change the test. I felt it was right to raise NotDvcRepoError rather than UrlNotDvcRepoError, as expected in the test. It just felt more correct to me. :)

Reply:
@skshetry which test though? :) Fixing code for the sake of test seems suspicious a bit, hence why I'm wondering.

Reply:
This is the test:
https://github.com/iterative/dvc/blob/ad6e58a45729c945e3e69bd9c949f8b27d55377e/tests/func/test_api.py#L54

Reply:
The only reason I used this is better wording:
`URL {} is not a DVC repository` vs worse `you are not inside of a repository`.

For local erepo, latter made more sense to me.

Reply:
@skshetry But for local repo it also doesn't make sense, as you might be `dvc get`ing from outside of the local repo, so that message becomes confusing.  `UrlNotDvcRepoError` is more appropriate there.

Reply:
Yes, the message is worse now then before. Just the `URLNotDvcRepoError` is imo better for url cases, and `NotDvcRepoError` for local cases (with external repo cases).

I was thinking of using `NotDvcRepoError` and changing message if it's for external repo in a different PR. But, I'll just rollback, as the change is confusing.

Reply:
If function is called `_require_dvc()` it should require dvc. 

@skshetry local repo is not always the one you are in, so it still should be `UrlNotDvcRepoError`.